components/input/input-otp.spec.ts

```diff
@@ -0,0 +1,233 @@
+    });
+    event.clipboardData?.setData('text', '789');
+    spyOn(event, 'preventDefault');
+
+    inputElements[0].triggerEventHandler('paste', event);
+    fixture.detectChanges();
+
+    expect(event.preventDefault).toHaveBeenCalled();
+    expect(component['otpArray'].at(0).value).toBe('*');
+    expect(component['internalValue'][0]).toBe('7');
+    expect(component['otpArray'].at(1).value).toBe('*');
+    expect(component['internalValue'][1]).toBe('8');
+    expect(component['otpArray'].at(2).value).toBe('*');
+    expect(component['internalValue'][2]).toBe('9');
+  });
+
+  it('should disable all inputs when setDisabledState is called with true', () => {
+    component.setDisabledState(true);
+    fixture.detectChanges();
+
+    inputElements.forEach(input => {
+      expect(input.nativeElement.disabled).toBe(true);
+    });
+  });
+
+  it('should recreate form array when nzLength changes', () => {
+    component.nzLength = 6;
+    component['createFormArray']();
+    const initialFormArray = component['otpArray'];
+
+    component.nzLength = 8;
+    component.ngOnChanges({
+      nzLength: {
+        currentValue: 8,
+        previousValue: 6,
+        firstChange: false,
+        isFirstChange: () => false
+      }
+    });
+
+    expect(component['otpArray']).not.toBe(initialFormArray);
+    expect(component['otpArray'].length).toBe(8);
+  });
+
+  it('should set disabled state correctly when disabled input changes', () => {
+    (component as NzSafeAny)['otpArray'] = new FormArray([]);
+    const spy = spyOn(component['otpArray'], 'disable').and.callThrough();
+    const enableSpy = spyOn(component['otpArray'], 'enable').and.callThrough();
+
+    component.disabled = true;
+    component.ngOnChanges({
+      disabled: {
+        currentValue: true,
+        previousValue: false,
+        firstChange: false,
+        isFirstChange: () => false
+      }
+    });
+
+    expect(spy).toHaveBeenCalled();
+    expect(enableSpy).not.toHaveBeenCalled();
+  });
+
+  it('should reset form array if the provided value is empty', () => {
+    const spy = spyOn(component['otpArray'], 'reset');
+
+    component.writeValue('');
+
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it('should set formatted values correctly in the form array', () => {
+    component.nzFormatter = value => value.toUpperCase();
+    component['createFormArray']();
+
+    component.writeValue('abcd');
+
+    expect(component['otpArray'].at(0).value).toBe('A');
+    expect(component['otpArray'].at(1).value).toBe('B');
+    expect(component['otpArray'].at(2).value).toBe('C');
+    expect(component['otpArray'].at(3).value).toBe('D');
+  });
+
+  it('should register onChange callback', () => {
+    const callback = jasmine.createSpy('onChangeCallback');
+    component.registerOnChange(callback);
+
+    component['onChangeCallback']?.('test value');
+
+    expect(callback).toHaveBeenCalledWith('test value');
+  });
+
+  it('should register onTouched callback', () => {
```

Comment:
These methods are implementation of `ControlValueAccessor` interface, which is well tested by angular. 
I think we needn't to write test cases for them, WDYT?

Reply:
You're absolutely right—logically, these parts shouldn't need to be tested again. However, for some reason, Karma can't recognize that they are part of Angular and expects us to test them. By removing these two tests, we would lose some code coverage. What’s your take on this?